scan/git.go

```diff
@@ -151,10 +151,15 @@ func checkoutGit(root, ref, subdir string) (string, error) {
 	// Try checking out by ref name first. This will work on branches and sets
 	// .git/HEAD to the current branch name
 	if output, err := gitWithinDir(root, "checkout", ref); err != nil {
-		// If checking out by branch name fails check out the last fetched ref
-		if _, err2 := gitWithinDir(root, "checkout", "FETCH_HEAD"); err2 != nil {
+		// If the branch name is specified, throw an error
```

Comment:
Why not allow specify the branch name?

Reply:
It means we try to checkout to that branch but fail. Therefore, we throw an error.